src/query/string_opfunc.c

```diff
@@ -3992,6 +3992,16 @@ db_string_pad (const MISC_OPERAND pad_operand, const DB_VALUE * src_string, cons
 			   db_get_string_length (src_string), db_get_string_size (src_string),
 			   db_get_string_codeset (src_string), &result, &result_type, &result_length, &result_size);
 
+  int max_size = MIN ((int) prm_get_bigint_value (PRM_ID_STRING_MAX_SIZE_BYTES),
```

Comment:
result_length = length in chars, should be compared to precision
result_size= size in bytes, should be compared with prm_get_bigint_value (PRM_ID_STRING_MAX_SIZE_BYTES)

Reply:
fixed